packages/dashmate/src/listr/tasks/setup/regular/importCoreDataTaskFactory.js

```diff
@@ -0,0 +1,221 @@
+import { Listr } from 'listr2';
+import fs from 'fs';
+import path from 'path';
+import {
+  NETWORK_TESTNET,
+} from '../../../../constants.js';
+
+/**
+ * @param {Config} config
+ * @return {validateCoreDataDirectoryPath}
+ */
+function validateCoreDataDirectoryPathFactory(config) {
+  /**
+   * @typedef {function} validateCoreDataDirectoryPath
+   * @param {string} value
+   * @return {string|boolean}
+   */
+  function validateCoreDataDirectoryPath(value) {
+    if (value.length === 0) {
+      return 'should not be empty';
+    }
+
+    // Path must be absolute
+    if (!path.isAbsolute(value)) {
+      return 'path must be absolute';
+    }
+
+    // Should contain data and dashd.conf
+    const configFilePath = path.join(value, 'dash.conf');
+
+    let dataDirName = 'blocks';
+    if (config.get('network') === NETWORK_TESTNET) {
+      dataDirName = 'testnet3';
+    }
+
+    try {
+      fs.accessSync(configFilePath, fs.constants.R_OK);
+      fs.accessSync(path.join(value, dataDirName), fs.constants.R_OK);
+    } catch (e) {
+      return 'directory must be readable and contain blockchain data with dash.conf';
+    }
+
+    const configFileContent = fs.readFileSync(configFilePath, 'utf8');
+
+    // Config file should contain testnet=1 in case of testnet
+    // and shouldn't contain testnet=1, regtest=1 or devnet= in case of mainnet
+    if (config.get('network') === NETWORK_TESTNET) {
+      if (!configFileContent.includes('testnet=1')) {
+        return 'dash.conf should be configured for testnet';
+      }
+    } else if (configFileContent.includes('testnet=1')
+      || configFileContent.includes('regtest=1')
+      || configFileContent.includes('devnet=')) {
+      return 'dash.conf should be configured for mainnet';
+    }
+
+    // Config file should contain masternodeblsprivkey in case of masternode
+    if (config.get('core.masternode.enable')) {
+      if (!configFileContent.includes('masternodeblsprivkey=')) {
+        return 'dash.conf should contain masternodeblsprivkey';
+      }
+    }
+
+    return true;
+  }
+
+  return validateCoreDataDirectoryPath;
+}
+
+/**
+ *
+ * @param {Docker} docker
+ * @param {dockerPull} dockerPull
+ * @param {generateEnvs} generateEnvs
+ * @return {importCoreDataTask}
+ */
+export default function importCoreDataTaskFactory(docker, dockerPull, generateEnvs) {
+  /**
+   * @typedef {function} importCoreDataTask
+   * @returns {Listr}
+   */
+  async function importCoreDataTask() {
+    return new Listr([
+      {
+        title: 'Import existing Core data',
+        task: async (ctx, task) => {
+          const doImport = await task.prompt({
+            type: 'toggle',
+            header: `  If you run a masternode on the same machine, you can
+   import your existing data so you don't need to sync node from scratch.
+   You current user must have read access to this directory.\n`,
```

Comment:
```suggestion
            header: `  If you already run a masternode on the machine, you can
   import your existing data instead of syncing from scratch.
   Your current user account must have read access to this directory.\n`,
```